docs/en/building/supplemental.md

```diff
@@ -0,0 +1,541 @@
+import 'a'; // removed
+'!has("bar")';
+import 'b'; // removed
+'!has("foo")';
+import 'c';
+'has("baz")';
+import 'd';
+```
+
+In the above example, imports `a` and `b` would be removed because their values evaluate to `true`. Import `c` will remain because it evaluates to `false`. Import `d` will remain because the `baz` feature has not been statically defined.
+
+## Known features
+
+The following features are recognized by the Dojo system and may be used internally to optimize code or shim features. If multiple features are specified then the intersection of available features will be returned.
+
+-   android
+-   chrome
+-   edge
+-   firefox
+-   ie11
+-   ios
+-   node
+-   node8
+-   safari
+
+## Provided Features
+
+These features are provided by the build system to help identify a specific environment or mode of operation.
+
+| Feature Flag        | Description                                                                                                                                                                                                                                              |
+| ------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `debug`             | Provides a way to create a code path for code that is only usable when debugging or providing enhanced diagnostics that are not desired in a _production_ build. Defaults to `true` but should be configured statically as `false` in production builds. |
+| `host-browser`      | Determines if the current environment contains a `window` and `document` object in the global context, therefore it is generally safe to assume the code is running in a browser environment.                                                            |
+| `host-node`         | Attempts to detect if the environment appears to be a node environment.                                                                                                                                                                                  |
+| `build-time-render` | Statically defined by the build-time rendering system during build-time rendering.                                                                                                                                                                       |
+
+# Build-time Rendering
+
+BTR renders a route to HTML during the build process and in-lines critical CSS and assets needed to display the initial view. This allows Dojo to pre-render the initial HTML used by a route and inject it directly into the page immediately. Skipping the initial render by creating it during the build process gives us many of the same benefits of server side rendering (SSR) such as performance gains and search engine optimization without the complexities of SSR.
```

Comment:
We don't "skip" the initial render, the application merges with the existing DOM structure during the initial render.

Reply:
I'm trying to say that the initial render of the DOM occurs during build time; during runtime it's appended to the DOM; after the application compares against that rather than generating the DOM from scratch.

Maybe "skip" the initial render isn't the right wording. How would you describe what's happening and why it improves performance?